### PR TITLE
[doc] Add an example for 'non-ascii-file-name'

### PR DESCRIPTION
**Pull request by Pierre-Sassoulas**
## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :scroll: Docs          |

## Description

Move links to related.rst, and upgrade the message shown for python 3.5+

Replace #8324, refs #5953

### CODE REVIEW

**Comment by ollie-iterators** [comment]
This was a tricky PR to work on. Good job on the pull request and I like that you used the file name "not_better", very funny.

File: pylint/checkers/non_ascii_names.py [hunk 1]
```diff
@@ -43,22 +43,15 @@ class NonAsciiNameChecker(base_checker.BaseChecker):
...
-                # Probably only a bug shows the problem quite good.
-                # That's also why this is a warning and not only a convention!
-                "Some editors don't support non-ASCII file names properly. "
-                "Even though Python supports UTF-8 files since Python 3.5 this isn't "
-                "recommended for interoperability. Further reading:\n"
-                "- https://peps.python.org/pep-0489/#export-hook-name\n"
-                "- https://peps.python.org/pep-0672/#confusing-features\n"
-                "- https://bugs.python.org/issue20485"
+                # files and had big troubles with git. Probably only a bug shows the problem
+                # quite good. That's also why this is a warning and not only a convention!
```
Comment (mbyrnepr2) [inline pylint/checkers/non_ascii_names.py:51]:
```suggestion
                # files. That's also why this is a warning and not only a convention!
```

For me, this content doesn't make the explanation clearer so perhaps we could remove or modify it.

File: pylint/checkers/non_ascii_names.py [hunk 2]
```diff
@@ -43,22 +43,15 @@ class NonAsciiNameChecker(base_checker.BaseChecker):
...
+                "Under python 3.5, PEP 3131 only allows non-ascii identifiers, not file names."
```
Comment (mbyrnepr2) [inline pylint/checkers/non_ascii_names.py:52]:
```suggestion
                "Under python 3.5, PEP 3131 allows non-ascii identifiers, but not non-ascii file names."
```
Perhaps a bit clearer 🤷‍♂️

File: doc/user_guide/configuration/all-options.rst [hunk 3]
```diff
@@ -202,6 +195,13 @@ Standard Checkers
 **Default:**  ``True``
 
 
+--source-roots
```
Comment (mbyrnepr2) [inline doc/user_guide/configuration/all-options.rst:198]: Is this meant to be part of this MR?

**Comment by mbyrnepr2** [comment]
I'm not sure the rationale behind the (empty) `good/bad.py` file or the `good/not_better.py` files?

**Review by mbyrnepr2 — COMMENTED** [review]
Just a few small comments :)

File: pylint/checkers/non_ascii_names.py (hunk 2 above)
Comment (Pierre-Sassoulas) [reply pylint/checkers/non_ascii_names.py:52]: The final whitespace will make the generated rst fail I'll edit before applying :)

File: doc/user_guide/configuration/all-options.rst (hunk 3 above)
Comment (Pierre-Sassoulas) [reply doc/user_guide/configuration/all-options.rst:198]: We forgot to generate the doc previousely because of https://github.com/PyCQA/pylint/pull/6736

**Comment by Pierre-Sassoulas** [comment]
> I'm not sure the rationale behind the (empty) good/bad.py file or the good/not_better.py files?

The problem is in the name of the file so we don't need any content imo. Maybe something like ``'à' is not an ascii character`` ?

**Comment by mbyrnepr2** [comment]
> > I'm not sure the rationale behind the (empty) good/bad.py file or the good/not_better.py files?
> 
> The problem is in the name of the file so we don't need any content imo. Maybe something like `"""'à' is not an ascii character but the file is named 'bàd.py'"""'` ?

I meant the file paths themselves. What does  good/bad.py mean? 😄

**Comment by Pierre-Sassoulas** [comment]
> I meant the file paths themselves. What does good/bad.py mean?

Ha, the doc framework render it as "correct code" vs "bad code" automatically see https://pylint--8340.org.readthedocs.build/en/8340/user_guide/messages/warning/non-ascii-file-name.html. There's two files because we assumed that a directory could be used only if there was multiple file to demonstrate and if there's only one file it need to be named bad.py.

**Comment by mbyrnepr2** [comment]
> > I meant the file paths themselves. What does good/bad.py mean?
> 
> Ha, the doc framework render it as "correct code" vs "bad code" automatically see https://pylint--8340.org.readthedocs.build/en/8340/user_guide/messages/warning/non-ascii-file-name.html. There's two files because we assumed that a directory could be used only if there was multiple file to demonstrate and if there's only one file it need to be named bad.py.

Awesome 😎

**Review by mbyrnepr2 — APPROVED** [review]

**Pull request merged**